dotcom-rendering/src/model/enhance-H2s.ts

```diff
@@ -11,13 +19,10 @@ const extractText = (element: SubheadingBlockElement): string => {
  * This function checks if the slug already exists and if it does it adds the count to the end of the slug.
  */
 const getUnique = (slug: string, array: string[]): string => {
-	if (array.includes(slug)) {
-		const occurenceCount = array.filter(
-			(currentItem) => currentItem === slug,
-		).length;
-		return `${slug}-${occurenceCount}`;
-	}
-	return slug;
+	const occurenceCount = array.filter(
+		(currentItem) => currentItem === slug,
+	).length;
+	return occurenceCount > 1 ? `${slug}-${occurenceCount - 1}` : slug;
```

Comment:
I think if it’s already in the array, you want to pad the slug.

```suggestion
	return occurenceCount > 0 ? `${slug}-${occurenceCount - 1}` : slug;
```

Reply:
+1

Reply:
This code does work but I agree it's confusing 😅 . The confusion is that the array of existingIds is already updated with the new id at this point (it happens in the calling function) so when it's counting occurrences it's also counting itself.

I've made a change to mutate that array _after_ the `getUnique` function is called

Reply:
Arg… I find this really hard to reason about.

Have we actually run some numbers on how often this double-id thing happens? It seems like we might be introducing a lot of complexity here for minimal gains.